compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1707,6 +1707,12 @@ crate fn show_candidates(
         candidates.iter().map(|c| path_names_to_string(&c.path)).collect();
 
     path_strings.sort();
+    let core_path_strings =
+        path_strings.iter().filter(|p| p.starts_with("core::")).cloned().collect::<Vec<String>>();
+    if !core_path_strings.is_empty() {
+        path_strings.retain(|p| !p.starts_with("core::"));
```

Comment:
This is minor, and we don't need to worry too much about the performance of generating suggestions, but it feels like this ought to be doable in one pass, removing and moving the elements matching `core::`.

This would be a good use case for the (currently unstable) `drain_filter` method.

Reply:
Thank you for your advice! I fixed it.